### DISCUSSION TITLE
sivercrest trv (Tuya TS0601(

### DISCUSSION THREAD

**Discussion by picasoaaa** [Koenkk/zigbee2mqtt#9676] [Devices]
the thermostatic head integrated as TS0601.

Why is he sending nonsense?

{
    "auto_lock": "MANUAL",
    "away_preset_days": 0,
    "away_preset_temperature": {
        "0": 6,
        "1": 34,
        "2": 24,
        "3": 42,
        "4": 36,
        "5": 34,
        "6": 68,
        "7": 42,
        "8": 92,
        "9": 34,
        "10": 96,
        "11": 42,
        "12": 96,
        "13": 34,
        "14": 96,
        "15": 42,
        "16": 96,
        "17": 34
    },
    "battery_low": {
        "0": 2,
        "1": 34,
        "2": 24,
        "3": 42,
        "4": 36,
        "5": 34,
        "6": 68,
        "7": 42,
        "8": 92,
        "9": 34,
        "10": 96,
        "11": 42,
        "12": 96,
        "13": 34,
        "14": 96,
        "15": 42,
        "16": 96,
        "17": 34
    },
    "boost_time": 42,
    "comfort_temperature": false,
    "current_heating_setpoint": 0,
    "holidays": [
        {
            "hour": 5,
            "minute": 34,
            "temperature": 24
        },
        {
            "hour": 42,
            "minute": 36,
            "temperature": 34
        },
        {
            "hour": 4,
            "minute": 42,
            "temperature": 92
        },
        {
            "hour": 34,
            "minute": 96,
            "temperature": 42
        },
        {
            "hour": 32,
            "minute": 34,
            "temperature": 96
        },
        {
            "hour": 42,
            "minute": 96,
            "temperature": 34
        }
    ],
    "linkquality": 84,
    "max_temperature": {
        "0": 0,
        "1": 0,
        "2": 0,
        "3": 30,
        "4": 80,
        "5": 92,
        "6": 34,
        "7": 96
    },
    "min_temperature": 10,
    "position": {
        "0": 1,
        "1": 34,
        "2": 24,
        "3": 42,
        "4": 36,
        "5": 34,
        "6": 68,
        "7": 42,
        "8": 92,
        "9": 34,
        "10": 96,
        "11": 42,
        "12": 96,
        "13": 34,
        "14": 96,
        "15": 42,
        "16": 96,
        "17": 34
    },
    "window_detection": "OFF",
    "window_detection_params": {},
    "workdays": [
        {
            "hour": 4,
            "minute": 34,
            "temperature": 24
        },
        {
            "hour": 42,
            "minute": 36,
            "temperature": 34
        },
        {
            "hour": 4,
            "minute": 42,
            "temperature": 92
        },
        {
            "hour": 34,
            "minute": 96,
            "temperature": 42
        },
        {
            "hour": 32,
            "minute": 34,
            "temperature": 96
        },
        {
            "hour": 42,
            "minute": 96,
            "temperature": 34
        }
    ]
}

**Comment by picasoaaa** [comment]
Silvercrest TVR01

**Reply by BerlinSnoop** [reply]
I have the same problem. 
But I am a Zigbee2mqt newbie and I am not sure if something is wrong configured. The (LIDL) Silvercrest is labeled with IAN: 368308_2010 and manufactured 06/2021. The Lidl HG06337 is working out of the box.

**Reply by MaartenVerb** [reply]
It is this device right?
https://www.lidl.nl/p/silvercrest-radiatorthermostaat-zigbee-smart-home/p100335076

I have it too and I also get the impression there is a mismatch between, what the device sends and what zigbee2mqtt expects.

**Comment by picasoaaa** [comment]
yes the device is correct

**Comment by picasoaaa** [comment]
how to ensure that the device sends the correct data?

**Reply by BerlinSnoop** [reply]
The solution is near here I think...
https://github.com/Koenkk/zigbee2mqtt/issues/9509
"Silvercrest/Lidl TS0601 - _TZE200_chyvmhay did not work but works with https://github.com/Lavisx/lidl-valve-z2m/blob/main/lidl_radiator_valve.js "

**Reply by MaartenVerb** [reply]
Yes, that helps. I'll test it better tomorrow.

**Reply by MaartenVerb** [reply]
Most of the functions do not seem to work. 
Battery 
Set temperature
current heating setpoint auto?

Functions that work;
Child lock
change mode to heat
Turn off, turn on.

![image](https://user-images.githubusercontent.com/88046766/143000825-6365bbaa-4dac-4f10-aefd-555bd676c212.png)

**Reply by MaartenVerb** [reply]
It does not seem to work well without the https://github.com/Lavisx/lidl-valve-z2m/blob/main/lidl_radiator_valve.js Zigbee2Mqtt keeps restarting then.

I've put the .js back, but I think features are not working or missing.

Will someone take another look at it?

**Reply by picasoaaa** [reply]
Tested today after last update core-2021.12.6, Zigbee2mqtt
Current version: 1.22.1-1
after steaming the head, zigbeetomqtt resets.
unusable in this state.

**Reply by gmarchart** [reply]
hi, i use the edge version of z2m without external converter - works very good. except 2 datapoints for summer/wintertime (automatic change) and window open detction (on/off) i see no issues at the moment.

Zigbee2MQTT version with deconz adapter
1.22.2-dev commit: 4f887a1

see #10540 
see #10601
